core/src/main/java/io/aiven/kafka/tieredstorage/RemoteStorageManager.java

```diff
@@ -164,7 +169,7 @@ private ObjectMapper getObjectMapper() {
             final SimpleModule simpleModule = new SimpleModule();
             simpleModule.addSerializer(SecretKey.class, new DataKeySerializer(rsaEncryptionProvider::encryptDataKey));
             simpleModule.addDeserializer(SecretKey.class, new DataKeyDeserializer(
-                b -> new SecretKeySpec(rsaEncryptionProvider.decryptDataKey(b), "AES")));
+                (keyId, b) -> new SecretKeySpec(rsaEncryptionProvider.decryptDataKey(keyId, b), "AES")));
```

Comment:
nit: wondering if we can move the `SecretKeySpec` definition into the `DataKeyDeserializer` 

```suggestion
            simpleModule.addDeserializer(SecretKey.class, new DataKeyDeserializer(rsaEncryptionProvider::decryptDataKey));
```

not directly related to the change, but we can take the chance to improve it.

`DataKeyDeserializer` would have a 

```
    private final BiFunction<String, byte[], byte[]> keyDecryptor;
```
instead, and return
```
        final var decryptedKey = keyDecryptor.apply(keyEncryptionKeyId, Base64.getDecoder().decode(dataKeyBase64));
        return new SecretKeySpec(decryptedKey, "AES");
```

Reply:
Yes, good idea. Done this, but also made `EncryptedDataKey` (ex-`DataEncryptionKeyResult`) passed between (de-)serializer and `RsaEncryptionProvider` both directions.